__tests__/app.test.js

```diff
@@ -40,4 +40,106 @@ describe('news-project', () => {
+        })
+
+        test('GET: 200 - a get request should return array of objects with a new property "comment_count"', () => {
+            return request(app).get('/api/articles').expect(200)
+            .then(({body: {articles}}) => {
+                expect(articles.length).toBeGreaterThan(0);
+                articles.forEach(article => {
+                    expect(article).toHaveProperty('comment_count', expect.any(String));
+                })
+            })
+        })
+
+        test('GET: 200 - a get request should return array of objects which is by default sorted by "created_at" in a DESC order', () => {
+            return request(app).get('/api/articles').expect(200)
+            .then(({body: {articles}}) => {
+                expect(articles[0]).toHaveProperty('created_at', '2020-11-03T09:12:00.000Z')
+                expect(articles[articles.length - 1]).toHaveProperty('created_at', '2020-01-07T14:08:00.000Z')
+            })
+        })
+    })
+
+    describe('/api/articles/:article_id/comments', () => {
+        test('GET: 200 - a get request should response with status 200', () => {
+            return request(app).get('/api/articles/1/comments').expect(200)
+        })
+
+        test('GET: 200 - a get request should return an array of comment objects', () => {
+            return request(app).get('/api/articles/1/comments').expect(200)
+            .then(({body: {comments}}) => {
+                expect(comments.length).toBeGreaterThan(0);
+                expect(comments).toBeInstanceOf(Array);
+                comments.forEach(comment => expect(comment).toBeInstanceOf(Object));
+            })
+        })
+
+        test('GET: 200 - a get request should return an array of objects with following properties', () => {
+            return request(app).get('/api/articles/1/comments').expect(200)
+            .then(({body: {comments}}) => {
+                comments.forEach(comment => 
+                    expect(comment).toEqual(expect.objectContaining({
+                        comment_id: expect.any(Number),
+                        votes: expect.any(Number),
+                        created_at: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                        article_id: expect.any(Number)
+                    }))
+                );
+            })
+        })
+
+        test('GET: 200 - a get request should return array of objects which is by default sorted by "created_at" in a DESC order', () => {
+            return request(app).get('/api/articles/1/comments').expect(200)
+            .then(({body: {comments}}) => {
+                expect(comments).toBeSortedBy('created_at', { descending: true })
+            })
+        })
+
+        test('GET: 404 - a get request should return a message "Not Found" if the article_id does not exist', () => {
+            return request(app).get('/api/articles/2/comments').expect(404)
```

Comment:
I am not sure that this test is worded correctly. Do you mean returns 404 when an article has no comments? There will be an article with an id of 2. I will leave a note in your models.js explaining this.

Reply:
Yeah, that there is no comments associated with article_id =2